src/extensions/agent_modules/shared/mosquitto_utils/src/adu_mosquitto_utils.c

```diff
@@ -140,7 +146,25 @@ bool ADU_are_correlation_ids_matching(
 
     if (outCorrelationData != NULL)
     {
-        if (ADUC_AllocAndStrCopyN(outCorrelationData, responseCorrelationId, (size_t)responseCorrelationIdLen) != 0)
+        // TODO: Remove this workaround once adu service deploys fix that ensures exact correlation id string agent sent is
+        // in the response message. Currently, it parses the UUID without dashes and then introduces dashes in the response
+        // message.
+        char correlation_id_no_dash[37] = "";
+        char* destEnd = NULL;
+        if (!ADUC_CopyIf(
```

Comment:
presumably this ADU's "C" version of std::copy_if ?

Reply:
Yes, basically.